app/v2/connectors/AuditingConnector.scala

```diff
@@ -71,4 +92,43 @@ class AuditingConnectorImpl @Inject() (httpClient: HttpClientV2, val metrics: Me
           .executeAndExpect(ACCEPTED)
     }
 
+  def post(
+    auditType: AuditType,
+    contentType: String,
+    contentLength: Long,
+    payload: Source[ByteString, _],
+    movementId: Option[MovementId] = None,
+    messageId: Option[MessageId] = None,
+    enrolmentEORI: Option[EORINumber] = None,
+    movementType: Option[MovementType] = None,
+    messageType: Option[MessageType] = None
+  )(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Unit] = withMetricsTimerAsync(MetricsKeys.AuditingBackend.Post) {
+    _ =>
+      val url = appConfig.auditingUrl.withPath(auditingRoute(auditType))
+      val path = hc.otherHeaders
+        .collectFirst {
+          case ("path", value) => value
+        }
+        .getOrElse("-")
+      httpClient
+        .post(url"$url")
+        .withInternalAuthToken
+        .setHeader(
+          HeaderNames.CONTENT_TYPE       -> contentType,
+          Constants.XContentLengthHeader -> contentLength.toString,
+          Constants.XMovementIdHeader    -> movementId.map(_.value).getOrElse(None).toString,
```

Comment:
```scala
@ None.toString
res1: String = "None"
```

This isn't what we want. If an element doesn't exist, the header shouldn't exist.

Reply:
Also, that `getOrElse(None)` is pointless. If the original option is `None`, `option.map(...)` is still `None`.